wd_helper.au3

```diff
@@ -652,27 +652,16 @@ EndFunc   ;==>_WD_FrameLeave
 ; Example .......: No
 ; ===============================================================================================================================
 Func _WD_HighlightElement($sSession, $sElement, $iMethod = Default)
-	Local Const $sFuncName = "_WD_HighlightElement"
-	Local Const $aMethod[] = _
-			[ _
-			"border: 0px", _
-			"border: 2px dotted red", _
-			"background: #FFFF66; border-radius: 5px; padding-left: 3px;", _
-			"border: 2px dotted red; background: #FFFF66; border-radius: 5px; padding-left: 3px;" _
-			]
-
-	If $iMethod = Default Then $iMethod = 1
-	If $iMethod < 0 Or $iMethod > 3 Then $iMethod = 1
+ 	Local Const $sFuncName = "_WD_HighlightElement"
 
-	Local $sScript = "arguments[0].style='" & $aMethod[$iMethod] & "'; return true;"
-	Local $sResult = _WD_ExecuteScript($sSession, $sScript,  __WD_JsonElement($sElement), Default, $_WD_JSON_Value)
+	Local $bResult = _WD_HighlightElements($sSession, $sElement, $iMethod)
 	Local $iErr = @error
 
 	If $_WD_DEBUG = $_WD_DEBUG_Info Then
```

Comment:
this is not needed as it is already done in `_WD_HighlightElements`
Thus 
```AutoIt
Local $iErr = @error 
```
is not needed just use:

```AutoIt
Return SetError($iErr, @error, $bResult)
```

Reply:
I don't think so, unless I've misunderstood what you are saying. You have to capture the @error coming out of `_WD_HighlightElements` and then pass it on to the calling routine.

Reply:
> You have to capture the @error coming out of `_WD_HighlightElements` and then pass it on to the calling routine.

Almost the same error will be notified firstly by `_WD_HighlightElements` and then again by `_WD_HighlightElement` so they will be duplicated/redundant.

Reply:
Especially since `_WD_HighlightElement` is just a wrapper for` _WD_HighlightElements`, and also because `_WD_HighlightElement` will be marked as Deprecated.

Reply:
Clearly we have a misunderstanding. Here are my observations --

- You can't eliminate `Local $iErr = @error` in _WD_HighlightElement without also changing the function's Return line. Otherwise, you get `warning: $iErr: possibly used before declaration.`

- The error code that is generated in `_WD_HighlightElements` needs to "bubble up" through `_WD_HighlightElement` so that the routine that initially called `_WD_HighlightElement` can check the value of @error to determine if an error occurred.

I understand that there will be some duplication as far as any logging to the console, but that isn't a concern for me.
Does that help? If not, what part of this doesn't make sense to you? Be specific. 😃

Reply:
To be specific, this is depracted function which is simple wrapper.
So I propose to truncate it as much as possible.:
```AutoIt
Func _WD_HighlightElement($sSession, $sElement, $iMethod = Default)
	Local $bResult = _WD_HighlightElements($sSession, $sElement, $iMethod)
	Return SetError(@error, @extended, $bResult)
EndFunc   ;==>_WD_HighlightElement
```

Reply:
I'm satisfied with the current revision. It's really a moot point since the function will eventually be removed.

Reply:
Ok.